### PR TITLE
DEVX-3607 Replace JWT generation with Nexmo Ruby JWT generator

### PR DESCRIPTION
**Pull request by hummusonrails**
This PR fulfills DEVX-3607 to replace the JWT generation in the SDK with the [Nexmo Ruby JWT](https://github.com/Nexmo/nexmo-jwt-ruby) generator.

### CODE REVIEW

**Review by mheap — CHANGES_REQUESTED** [review]
Can we also have a test when `ttl` is passed? I know it's part of the JWT lib tests, but it's also a part of the SDKs public API as we expose it through `Nexmo::JWT`

I'd also like to see `Time.now` mocked and a complete JWT string comparison done as the tests here don't validate that the generated JWT is valid

File: lib/nexmo/jwt.rb [hunk 1]
```diff
@@ -33,15 +33,9 @@ module JWT
...
-    def self.generate(payload, private_key)
-      payload[:iat] = iat = Time.now.to_i unless payload.key?(:iat) || payload.key?('iat')
-      payload[:exp] = T.must(iat) + 60 unless payload.key?(:exp) || payload.key?('exp')
-      payload[:jti] = SecureRandom.uuid unless payload.key?(:jti) || payload.key?('jti')
-
-      private_key = OpenSSL::PKey::RSA.new(private_key) unless private_key.respond_to?(:sign)
-
-      ::JWT.encode(payload, private_key, 'RS256')
+    def self.generate(payload, private_key = nil)
+      payload[:private_key] = private_key unless payload[:private_key]
```
Comment (mheap) [inline lib/nexmo/jwt.rb]: Should this be `if private_key && !payload[:private_key]`?

Comment (hummusonrails) [reply lib/nexmo/jwt.rb]: You're right! What about also adding a `raise` as well if neither are present?

File: lib/nexmo/jwt.rb [hunk 2]
```diff
@@ -33,15 +33,11 @@ module JWT
...
     #
-    sig { params(payload: T::Hash[T.any(Symbol, String), T.any(String, Integer)], private_key: T.any(OpenSSL::PKey::RSA, String)).returns(String) }
-    def self.generate(payload, private_key)
-      payload[:iat] = iat = Time.now.to_i unless payload.key?(:iat) || payload.key?('iat')
-      payload[:exp] = T.must(iat) + 60 unless payload.key?(:exp) || payload.key?('exp')
-      payload[:jti] = SecureRandom.uuid unless payload.key?(:jti) || payload.key?('jti')
-
-      private_key = OpenSSL::PKey::RSA.new(private_key) unless private_key.respond_to?(:sign)
+    def self.generate(payload, private_key = nil)
+      raise "Expecting 'private_key' in either the payload or as a separate parameter" if !payload[:private_key] && !private_key
```
Comment (mheap) [inline lib/nexmo/jwt.rb:37]: Missing a test for this branch

**Review by mheap — CHANGES_REQUESTED** [review]

File: lib/nexmo/jwt.rb (hunk 2 above)
Comment (hummusonrails) [reply lib/nexmo/jwt.rb:37]: Good catch!

**Pull request merged**